### PR TITLE
fix php_module_startup call for PHP 8.2

### PR DESCRIPTION
**Pull request by remicollet**
Without this patch

```
+ : PHP module configuration
+ ./configure php --config=/opt/remi/php82/root/usr/bin/php-config --module=php82
configuring PHP module
checking for PHP ... found
 + PHP SAPI: [apache2handler litespeed fpm phpdbg cli embed cgi]
checking for PHP version ... 8.2.0-dev
checking for PHP embed SAPI ... not found

./configure: error: no PHP embed SAPI found.

```

### CODE REVIEW

**Comment by tippexs** [comment]
Thanks @remicollet for pointing this out! Did you have a link to the C API changelog just for documentation? That would be perfect!

**Comment by alejandro-colomar** [comment]
@artemkonev We should add a changelog for this, right?  Supporting PHP 8.2

**Comment by remicollet** [comment]
So, https://github.com/php/php-src/blob/master/UPGRADING.INTERNALS#L65

**Review by alejandro-colomar — APPROVED** [review]

**Comment by alejandro-colomar** [comment]
> > Did you have a link to the C API changelog just for documentation?
> 
> Sorry nothing in UPGRADINGS (perhaps @Girgias may help, as he changes it in [php/php-src@b5db594fd27](https://github.com/php/php-src/commit/b5db594fd27))
> 
> > @artemkonev We should add a changelog for this, right? Supporting PHP 8.2
> 
> Probably too early, **alpha1** will be announced tomorrow, API is not yet final, better to wait for RC and final API

Okay, so please ping when you have an RC.  I'll leave this open for now.

**Comment by remicollet** [comment]
> Okay, so please ping when you have an RC. I'll leave this open for now.

You can merge the "code" fix without updating doc claiming it is 8.2 compatible ;)

**Comment by alejandro-colomar** [comment]
> > Okay, so please ping when you have an RC. I'll leave this open for now.
> 
> You can merge the "code" fix without updating the "doc" claiming it is 8.2 compatible ;)

That also makes sense.

@artemkonev What do we say in the changelog?

> 
> P.S. obviously I will forget the ping....

To be expected :)

**Comment by andypost** [comment]
PHP 8.2.0 RC4 released, using this patch to build module - it works perfect, please commit the fix

File: auto/modules/php [hunk 1]
```diff
@@ -149,7 +149,11 @@ nxt_feature_test="
     #include <php_main.h>
 
     int main() {
+    #if PHP_VERSION_ID < 80200
```
Comment (ac000) [inline auto/modules/php:152]: The check should be in brackets (to match Unit's coding style). I.e

```diff
#if (PHP_VERSION_ID < 80200)
```

File: src/nxt_php_sapi.c [hunk 2]
```diff
@@ -1150,7 +1150,11 @@ nxt_php_vcwd_chdir(nxt_unit_request_info_t *req, u_char *dir)
 static int
 nxt_php_startup(sapi_module_struct *sapi_module)
 {
+#if PHP_VERSION_ID < 80200
```
Comment (ac000) [inline src/nxt_php_sapi.c:1153]: Same here...

**Review by ac000 — CHANGES_REQUESTED** [review]

File: auto/modules/php (hunk 1 above)
Comment (remicollet) [reply auto/modules/php]: brackets added (despite there is some other places without ex https://github.com/nginx/unit/blob/master/src/nxt_php_sapi.c#L19)

**Comment by alejandro-colomar** [comment]
Looks good to me.  I think we can already add the changelog now, since I wouldn't expect big breaking changes after the RC.

**Comment by ac000** [comment]
Heh, I think they can be squashed into a single commit...

**Comment by alejandro-colomar** [comment]
> Heh, I think they can be squashed into a single commit...

Actually I'd squash them into 2.  The fixes for existing style I'd keep them separate.  I can do that at the moment of applying, so no problem.

**Comment by ac000** [comment]
You mean squash the two brackets patches into one?

**Comment by alejandro-colomar** [comment]
> You mean squash the two brackets patches into one?

I'd do a selective rebase so that one of the brackets and part of the other are squasehd into the main commit, and then part of one of the brakcet commits still survives as separate.

**Comment by ac000** [comment]
Ah, OK I see now, the 'add more brackets' patch is touching a lot of 'other' code, I did wonder why there was two 'bracket' patches...

**Review by ac000 — APPROVED** [review]

**Comment by alejandro-colomar** [comment]
Hi!

I rebased to master, rewrote the commit logs,and signed on your behalf:

```git
$ git log -2
commit 370e290c37fc2490fb4e02512a3ce3a8698ffe3b (HEAD -> php)
Author: Remi Collet <remi@remirepo.net>
Date:   Fri Oct 14 12:16:43 2022 +0200

    Added parentheses for consistency
    
    Reported-by: Andrew Clayton <a.clayton@nginx.com>
    Signed-off-by: Remi Collet <remi@remirepo.net>
    Reviewed-by: Andrew Clayton <a.clayton@nginx.com>
    Signed-off-by: Alejandro Colomar <alx@nginx.com>

commit 8b4fe01e0663f3d6418bcdaf195bf72bca56cea8
Author: Remi Collet <remi@remirepo.net>
Date:   Thu Jun 2 16:16:35 2022 +0200

    PHP: Fixed php_module_startup() call for PHP 8.2
    
    PHP 8.2 changed the prototype of the function, removing the last
    parameter.
    
    Signed-off-by: Remi Collet <remi@remirepo.net>
    Cc: Timo Stark <t.stark@nginx.com>
    Cc: George Peter Banyard <girgias@php.net>
    Tested-by: Andy Postnikov <apostnikov@gmail.com>
    Acked-by: Andy Pstnikov <apostnikov@gmail.com>
    Reviewed-by: Andrew Clayton <a.clayton@nginx.com>
    Signed-off-by: Alejandro Colomar <alx@nginx.com>
```

And added the following changelog:

```diff
diff --git a/docs/changes.xml b/docs/changes.xml
index 5da00d88..96877044 100644
--- a/docs/changes.xml
+++ b/docs/changes.xml
@@ -31,6 +31,12 @@ NGINX Unit updated to 1.29.0.
          date="" time=""
          packager="Nginx Packaging &lt;nginx-packaging@f5.com&gt;">
 
+<change type="feature">
+<para>
+compatibility with PHP 8.2.
+</para>
+</change>
+
 <change type="feature">
 <para>
 support rack v3 in ruby applications.

```

I'll push this in a moment.

**Comment by alejandro-colomar** [comment]
Merged.

**Comment by Vrtak-CZ** [comment]
Hi, are there any plans for releasing (as 1.28.1) this prior to PHP 8.2.0 stable release (planned for December 8th 2022)?

**Comment by Vrtak-CZ** [comment]
Hi guys, PHP 8.2.0 has been released. Are there any plans for releasing a new version of Nginx Unit with this fix?

**Comment by alejandro-colomar** [comment]
Hi!

Yes, if all goes well, we'll release in a week or so.

Cheers,
Alex

**Pull request closed without merging**